cmd/template/nodepool/provider/aws.go

```diff
@@ -64,3 +149,59 @@ func WriteAWSTemplate(out io.Writer, config NodePoolCRsConfig) error {
 
 	return nil
 }
+
+func getCAPANodepoolTemplate(config NodePoolCRsConfig) (client.Template, error) {
+	var err error
+
+	c, err := client.New("")
+	if err != nil {
+		return nil, err
+	}
+
+	templateOptions := client.GetClusterTemplateOptions{
+		ClusterName:       config.ClusterID,
+		TargetNamespace:   config.Owner,
+		KubernetesVersion: "v1.19.9",
```

Comment:
We need to find a way to get the desired kubernetes version here. If it should come from the release CR, we need to default it later. However, the request will not get through when this here is not set. We can also consider getting it from a flag, but that would open up customers to use potentially untested versions.

Reply:
Can't we set it in the admission controller, based on the release version?

Reply:
Unfortunately not. If this is not set in the `templateOptions` the request is blocked and we do not get a template that could be defaulted by the admission-controller.